src/driver/safe/external_memory.rs

```diff
@@ -0,0 +1,124 @@
+use core::{marker::PhantomData, mem::ManuallyDrop};
+use std::fs::File;
+use std::ops::Range;
+use std::sync::Arc;
+
+use super::{CudaDevice, DevicePtr, DeviceSlice};
+use crate::driver::{result, sys, DriverError};
+
+impl CudaDevice {
+    /// Import external memory from a [`File`].
+    ///
+    /// # Safety
+    /// `size` must be the size of the external memory.
+    #[cfg(any(unix, windows))]
+    pub unsafe fn import_external_memory(
+        self: &Arc<Self>,
+        file: File,
+        size: u64,
```

Comment:
Could you use `file`'s metadata for this? https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.len

Reply:
I experimented with this, and the `File` objects I get [from Vulkan](https://docs.rs/vulkano/latest/vulkano/memory/struct.DeviceMemory.html#method.export_fd) do not fill the metadata. When I looked at the length it was 0.
I could just use the metadata and enforce that the caller sets it correctly, but that seemed more error-prone than asking for it explicitly like this.

Reply:
Oh weird, okay. Yeah I think that's fine then